### PR TITLE
fixed object locking mechanism in s3

### PR DESCRIPTION
**Pull request by macnev2013**
## Fixed
object locking mechanism in s3

## Added
S3_OBJECT_LEGAL_HOLD 

## Test Passing
TestAccAWSS3Bucket_forceDestroyWithObjectLockEnabled

## Note
This is a test PR. Please do not merge as of now.

### CODE REVIEW

File: moto/s3/responses.py
Comment (whummer) [inline moto/s3/responses.py:835]: nit: The true strings could be put into a re-usable global constant like `TRUE_STRINGS = ["true", "True"]`. Perhaps such a constant already exists in the codebase @bblommers ?

**Review by whummer — APPROVED** [review]
LGTM, only two minor remarks - thanks for adding this @macnev2013 .

Waiting for review from @bblommers .. 🙏 Thanks

File: moto/s3/responses.py
Comment (bblommers) [reply moto/s3/responses.py:1588]: Agreed, that's much easier to read IMO

File: moto/s3/responses.py
Comment (bblommers) [reply moto/s3/responses.py:835]: In other places we tend to solve this with `x.lower() == "true"`, so for consistency sake, would be good to have the same here

**Comment by whummer** [comment]
> @whummer @bblommers, This PR will have some breaking effect on localstack. Some feature could break due to this. As a fix i have created a new PR in localstack here.

Thanks @macnev2013 - should be good to merge here. 👍 (we'll take care of merging / resolving on LocalStack side on our end then..). Thanks @bblommers

**Pull request merged**
